### PR TITLE
Langchain Multihop as an experimental feature

### PR DESCRIPTION
**Pull request by KevinLuong96**
- [ ] **PR title**: "area: description"
  - Where "area" is whichever of interface, frontend, model, tools, backend, etc. is being modified. Use "docs: ..." for purely docs changes, "infra: ..." for CI changes.
  - Example: "deployment: add Azure model option"

Description: Currently, multihop is only supported through use of Langchain ReAct agents. We want to allow users to set an environment variable to enable this experimental feature.

This PR:
- Creates an `/experimental_features` endpoint which returns a dictionary of experimental features and their state
- Adds Storage to the langchain chat endpoint
- Returns `Langchain_tools` as available tools when the flag is set to True
- Updates the langchain chat flow to use the generic chat helper functions and 

- [ ] **Add tests and docs**: Please include testing and documentation for your changes
- [ ] **Lint and test**: Run `make lint` and `make test`

### CODE REVIEW

File: src/backend/chat/custom/langchain.py [hunk 1]
```diff
@@ -25,43 +39,27 @@ def __init__(self) -> None:
...
-        ].implementation()
+        tools = []
+        for req_tool in chat_request.tools:
+            tool = AVAILABLE_TOOLS.get(req_tool.name)
+            if tool:
+                tools.append(tool.implementation().to_langchain_tool())
+            else:
+                raise ValueError(f"Tool {req_tool.name} not found")
+
         # Create the ReAct agent
```
Comment (scott-cohere) [inline src/backend/chat/custom/langchain.py:50]:
```suggestion
        # Create the React agent
```

File: src/backend/routers/chat.py [hunk 2]
```diff
@@ -658,19 +666,50 @@ def generate_chat_response(
 
 
 @router.post("/langchain")
```
Comment (scott-cohere) [inline src/backend/routers/chat.py:668]: nit: maybe `/langchain-chat` or something

File: src/backend/schemas/langchain_chat.py [hunk 3]
```diff
@@ -1,23 +1,9 @@
...
-    preamble: str | None = Field(
-        default=None,
-        title="A string to override the preamble.",
-    )
-    tools: List[Tool] = Field(
-        default_factory=list,
-        title="List of managed tools to use for the response.",
-        exclude=True,
-    )
+    pass
```
Comment (scott-cohere) [inline src/backend/schemas/langchain_chat.py:9]: hmmm whats the point of this? just to differentiate req types for diff chat reqs?

**Review by scott-cohere — APPROVED** [review]

File: src/backend/routers/chat.py [hunk 4]
```diff
@@ -753,11 +790,9 @@ async def generate_langchain_chat_stream(
...
             if event.get("output", "") and event.get("citations", []):
-                # TODO: link citations to results from tool steps
+                final_message_text = event.get("output", "")
                 stream_event = StreamEnd(
-                    response_id=str(uuid4()),  # TODO make this optional
-                    generation_id=str(uuid4()),  # TODO make this optional
-                    conversation_id=chat_request.conversation_id,
+                    conversation_id=conversation_id,
                     text=event.get("output", ""),
                     citations=[],
```
Comment (scott-cohere) [inline src/backend/routers/chat.py:797]: uhh should this be `citations=event.get("citation", "")` ?

File: src/backend/routers/chat.py (hunk 2 above)
Comment (scott-cohere) [reply src/backend/routers/chat.py:668]: oh nvm citations not supported.. maybe leave comment here too?

File: src/backend/routers/chat.py (hunk 4 above)
Comment (scott-cohere) [reply src/backend/routers/chat.py:797]: oh nvm citations not supported.. maybe leave comment here too?

File: src/backend/schemas/langchain_chat.py (hunk 3 above)
Comment (KevinLuong96) [reply src/backend/schemas/langchain_chat.py:9]: yeah, there's a lot of fields on cohere chat request which are not required. Better to only pass around what is needed IMO

File: src/backend/chat/custom/langchain.py (hunk 1 above)
Comment (KevinLuong96) [reply src/backend/chat/custom/langchain.py:49]: It's actually ReAct haha

Comment (KevinLuong96) [reply src/backend/chat/custom/langchain.py:49]: https://python.langchain.com/docs/modules/agents/agent_types/react/

**Review by scott-cohere — APPROVED** [review]

**Pull request merged**